arbor/include/arbor/gpu/gpu_common.hpp

```diff
@@ -30,7 +33,36 @@ constexpr inline unsigned block_count(unsigned n, unsigned block_size) {
     return (n+block_size-1)/block_size;
 }
 
+inline void device_error(const api_error_type& api_error, const char func[], const char file[], int line) {
+    std::ostringstream o;
+    o << "device error: \"" << api_error.description() << "\" [" << api_error.name() << "]"
+      << " in function: " << func << ", location: " << file << ":" << line;
+    throw std::runtime_error(o.str());
+}
+
 } // namespace impl
 
+#define ARB_GPU_CHECK(api_error)                          \
```

Comment:
This would be helpful in `synchronize` as well, no?

Reply:
where would you put it exactly? we do not use `synchronize` apart from just below and in the profiler (meters) - I could also add the macro in the latter code if you want?

Reply:
ok added check in gpu_context